platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -477,22 +479,61 @@ protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean
         return new LDAPConnectionFactory(host, port, lo);
     }
 
-    /**
-     * Added additional logging to the security LOGGER.
-     */
-    @Override
-    public boolean login() throws LoginException {
-        try {
-            boolean isLoggedIn = doLogin();
-
-            if (!isLoggedIn) {
-                SecurityLogger.audit("Username [" + user + "] failed LDAP authentication.");
+    void validateUsername(String username) throws LoginException {
+        boolean hasBadCharacters = false;
+        for (int i = 0; i < username.length(); i++) {
+            char curChar = username.charAt(i);
+            switch (curChar) {
+            case '\\':
```

Comment:
Yeah, I did this because it was the most performant way of completing the check and it is the OWASP recommendation. We could change it to a whitelist instead, but there are a lot more characters that we'd need to potentially allow. You could use a regex for most of them, but that will slow things down a bit. I'm good with either, if you want to change it. Whitelisting is certainly a security best practice (as long as we don't overly restrict it) and what is already here is the OWASP recommendation, so either is good with me.